File: cpp/1st_Week/MariMencoba/Balok.cpp
#include <iostream>
#include <string>

using namespace std;

class Balok
{
    //properties balok
    private:
        int panjang;
        int lebar;
        int tinggi;
    
    public:


    //constructor
    Balok(){
    }

    Balok(int p, int l, int t){
        panjang = p;
        lebar = l;
        tinggi = t;
    }

    //method get
    int getPanjang(){ return panjang;}

    int getLebar(){ return lebar; }

    int getTinggi(){ return tinggi; }

    //method set
    void setPanjang(int p){ panjang = p; }

    void setLebar(int l){ lebar = l; }

    void setTinggi(int t){ tinggi = t; }

    //method volume
    int volume(){
        return panjang*lebar*tinggi;
    }

    int luasPermukaan(){
        return (2*panjang*lebar)+(2*lebar*tinggi)+(2*tinggi*panjang);
    }

    //destructor
    ~Balok(){
    }
};

File: python/1st_week/MariMencoba/Balok.py
class Balok:
    # class unutk balok

    def __init__(self, panjang, lebar, tinggi):
        self.panjang = panjang
        self.lebar = lebar
        self.tinggi = tinggi

    # prosedur get
    def getPanjang(self):
        return self.panjang
    
    def getLebar(self):
        return self.lebar

    def getTinggi(self):
        return self.tinggi

    # fungsi set
    def setPanjang(self, p):
        self.panjang = p
    
    def setLebar(self, l):
        self.lebar = l
    
    def setTinggi(self, t):
        self.tinggi = t

    # fungsi volume
    def volume(self):
        return self.panjang*self.lebar*self.tinggi
    
    # fungsi luas permukaan
    def luasPermukaan(self):
        return (2*self.panjang*self.lebar)+(2*self.lebar*self.tinggi)+(2*self.tinggi*self.panjang)